README needs to be updated - Makefile image and push targets does not exist

### What happened?

The Readme is referring to `make image` and `make push` but when using them an error is returned:

```
make: *** No rule to make target `image'.  Stop.
make: *** No rule to make target `push'.  Stop.
```

### How can we reproduce it?

```
make image
```

### What environment did it happen in?

```
➜  provider-jet-mything git:(master) ✗ make submodules
➜  provider-jet-mything git:(master) ✗ make   
20:42:40 [ .. ] verify dependencies have expected content
all modules verified
20:43:07 [ OK ] go modules dependencies verified
20:43:07 [ .. ] golangci-lint
20:43:24 [ OK ] golangci-lint
20:43:25 [ .. ] go build linux_amd64
go build: -i flag is deprecated
github.com/crossplane-contrib/provider-jet-mything/apis/v1alpha1
github.com/crossplane-contrib/provider-jet-mything/config
github.com/crossplane-contrib/provider-jet-mything/apis/hashicups/v1alpha1
github.com/crossplane-contrib/provider-jet-mything/internal/clients
github.com/crossplane-contrib/provider-jet-mything/internal/controller/providerconfig
github.com/crossplane-contrib/provider-jet-mything/apis
github.com/crossplane-contrib/provider-jet-mything/internal/controller/hashicups/order
github.com/crossplane-contrib/provider-jet-mything/internal/controller
github.com/crossplane-contrib/provider-jet-mything/cmd/provider
20:43:32 [ OK ] go build linux_amd64
20:43:34 [ .. ] docker build build-adcc6e90/provider-jet-mything-amd64
sha256:875a8420addfe828899db7337f3c28343bd67d1cbdf26ab8feefdbb7b7f4ce45
20:43:35 [ OK ] docker build build-adcc6e90/provider-jet-mything-amd64
20:43:37 [ .. ] docker build build-adcc6e90/provider-jet-mything-controller-amd64
sha256:03ee46aed4e2e9583616f320e9addae6966d7753b62d96e1f7bc2c943749353a
20:43:43 [ OK ] docker build build-adcc6e90/provider-jet-mything-controller-amd64
➜  provider-jet-mything git:(master) ✗ make image
make: *** No rule to make target `image'.  Stop.
➜  provider-jet-mything git:(master) ✗ make push
make: *** No rule to make target `push'.  Stop.
➜  provider-jet-mything git:(master) ✗ 
```

--- a/README.md
+++ b/README.md
@@ -34,18 +34,6 @@ Build, push, and install:
 make all
 ```
 
-Build image:
-
-```console
-make image
-```
-
-Push image:
-
-```console
-make push
-```
-
 Build binary:
 
 ```console

